server/services/echo_service.go

```diff
@@ -98,6 +99,106 @@ func (s *echoServerImpl) EchoErrorDetails(ctx context.Context, in *pb.EchoErrorD
 	return response, nil
 }
 
+// DetailedError satisfies the interface defined in https://pkg.go.dev/google.golang.org/grpc/status#FromError to convert an error to a status.Status.
+type DetailedError struct {
+	grpcStatus *status.Status
+}
+
+func (de DetailedError) Error() string {
+	return fmt.Sprintf("DetailedError.Error(): %d: %s [(%d details)]", de.grpcStatus.Code(), de.grpcStatus.Message(), len(de.grpcStatus.Details()))
+}
+
+// GRPCStatus returns the gRPC status associated with the given
+// DetailedError as a way of satisfying the interface defined in
+// https://pkg.go.dev/google.golang.org/grpc/status#FromError
+func (de *DetailedError) GRPCStatus() *status.Status {
+	return de.grpcStatus
+}
+
+func (s *echoServerImpl) FailEchoWithDetails(ctx context.Context, in *pb.FailEchoWithDetailsRequest) (*pb.FailEchoWithDetailsResponse, error) {
```

Comment:
So this is intended to return all of the standard error details so that clients can exercise parsing/be sure that all of them work?

The RPC documents that a bit, but would you mind including that on the function comment as well? Or I guess extend the proto comment a bit more?

Reply:
Done, WDYT?